### PR TITLE
Basic PHP plugin

### PR DESCRIPTION
**Pull request by kaloyan-raev**
### What does this PR do?

This is the first step of providing better PHP language support in Che.
This basic plugin is created after the existing Python plugin and
provides the PHP project type and the "New PHP File" action.
### What issues does this PR fix or reference?
1. Adds the PHP project type.
2. Adds the New PHP File action.
### New Behavior

This content may also be included in the release notes.
### Tests written?

No
### Docs requirements?

Include the content for all the docs changes required. 
1.  User docs changes
- https://eclipse-che.readme.io/docs/workspace

Please review [Che's Contributing Guide](https://github.com/eclipse/che/CONTRIBUTING.MD) for best practices.

### CODE REVIEW

File: plugins/plugin-php/che-plugin-php-lang-ide/src/main/java/org/eclipse/che/plugin/php/ide/PhpExtension.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+/**
+ * @author Kaloyan Raev
+ */
+@Extension(title = "PHP")
+public class PhpExtension {
+
+    public static String PHP_CATEGORY = "PHP";
+
+    @Inject
+    public PhpExtension(FileTypeRegistry fileTypeRegistry,
```
Comment (vparfonov) [inline plugins/plugin-php/che-plugin-php-lang-ide/src/main/java/org/eclipse/che/plugin/php/ide/PhpExtension.java:37]: formatting

File: plugins/plugin-php/che-plugin-php-lang-ide/src/main/java/org/eclipse/che/plugin/php/ide/action/CreatePhpSourceFileAction.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+@Singleton
+public class CreatePhpSourceFileAction extends NewPhplikeResourceAction {
+
+
+    private static final String DEFAULT_CONTENT = "<?php\n" +
+                                                  "\n";
+
+    @Inject
+    public CreatePhpSourceFileAction(PhpLocalizationConstant localizationConstant,
+                                        PhpResources resources,
```
Comment (vparfonov) [inline plugins/plugin-php/che-plugin-php-lang-ide/src/main/java/org/eclipse/che/plugin/php/ide/action/CreatePhpSourceFileAction.java]: formatting

File: plugins/plugin-php/che-plugin-php-lang-ide/src/main/java/org/eclipse/che/plugin/php/ide/action/NewPhplikeResourceAction.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+     * Creates new action.
+     *
+     * @param title
+     *         action's title
+     * @param description
+     *         action's description
+     * @param svgIcon
+     */
+    public NewPhplikeResourceAction(String title,
+                                       String description,
```
Comment (vparfonov) [inline plugins/plugin-php/che-plugin-php-lang-ide/src/main/java/org/eclipse/che/plugin/php/ide/action/NewPhplikeResourceAction.java]: formatting

**Review by vparfonov — COMMENTED** [review]
@kaloyan-raev fix formatting please an i will merge it

File: plugins/plugin-php/che-plugin-php-lang-ide/src/main/java/org/eclipse/che/plugin/php/ide/PhpExtension.java (hunk 1 above)
Comment (kaloyan-raev) [reply plugins/plugin-php/che-plugin-php-lang-ide/src/main/java/org/eclipse/che/plugin/php/ide/PhpExtension.java:37]: Done

File: plugins/plugin-php/che-plugin-php-lang-ide/src/main/java/org/eclipse/che/plugin/php/ide/action/CreatePhpSourceFileAction.java (hunk 2 above)
Comment (kaloyan-raev) [reply plugins/plugin-php/che-plugin-php-lang-ide/src/main/java/org/eclipse/che/plugin/php/ide/action/CreatePhpSourceFileAction.java]: Done

File: plugins/plugin-php/che-plugin-php-lang-ide/src/main/java/org/eclipse/che/plugin/php/ide/action/NewPhplikeResourceAction.java (hunk 3 above)
Comment (kaloyan-raev) [reply plugins/plugin-php/che-plugin-php-lang-ide/src/main/java/org/eclipse/che/plugin/php/ide/action/NewPhplikeResourceAction.java]: Done

**Pull request merged**